Firestore/core/test/firebase/firestore/model/transform_operations_test.cc

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Firestore/core/src/firebase/firestore/model/transform_operations.h"
+
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace model {
+
+class DummyOperation : public TransformOperation {
+ public:
+  DummyOperation() {
+  }
+
+  Type type() const override {
+    return static_cast<Type>(-1);
+  }
+
+  bool operator==(const TransformOperation* other) const override {
+    return this == other;
+  }
+};
+
+TEST(TransformOperations, ServerTimestamp) {
+  ServerTimestampTransform transform = ServerTimestampTransform::Get();
+  EXPECT_EQ(TransformOperation::Type::ServerTimestamp, transform.type());
+
+  ServerTimestampTransform another = ServerTimestampTransform::Get();
+  DummyOperation dummy;
+  EXPECT_TRUE(transform.operator==(&another));
```

Comment:
How about:
`EXPECT_EQ(transform, another)`

Or if you really want to allow comparing an object to a pointer:
`EXPECT_EQ(transform, &another)`

(similar for following line)